sound/soc/sof/sof-audio.c

```diff
@@ -270,14 +270,16 @@ sof_unprepare_widgets_in_path(struct snd_sof_dev *sdev, struct snd_soc_dapm_widg
 	struct snd_sof_widget *swidget = widget->dobj.private;
 	struct snd_soc_dapm_path *p;
 
-	if (!widget_ops[widget->id].ipc_unprepare || !swidget->prepared)
-		goto sink_unprepare;
+	/* no need to check sink */
```

Comment:
@RanderWang this is a new change right? Im just tryng to understand why we dont have to check the sink if the source is unprepared. Are we saying: if the source is unprepared the sink must also be unprepared?

Reply:
sure. If the source is unprepared, the sink has been unprepared or is being unprepared (for loop case), so no need to unprepare it.

Reply:
if there is a loop case,  like s0->s1->s2->......>s1, it will result to infinite loop without check unprepare == false.